Kickstarter-iOS/ViewModels/RootViewModelTests.swift

```diff
@@ -280,6 +280,28 @@ final class RootViewModelTests: TestCase {
 
     self.tabBarItemsData.assertValues([tabData, tabDataLoggedIn, tabDataLoggedIn, tabData, tabDataMember])
   }
+
+  func testSetViewControllers_DoesNotFilterDiscovery() {
```

Comment:
this test wouldn't have failed with the previous code unfortunately.

i would suggest reverting the vm to go back to `combineLatest` and then get a failing test that the `takeWhen` fixes.

essentially what we are testing is:

* view loads and `filterDiscovery` doesnt emit
* we call `switchToDiscovery` explicitly (cause some other part of the app wanted to), and that causes `filterDiscovery` to emit
* later a user logs in (or out) causing the controllers to be reloaded, _but_ `filterDiscovery` does not emit again